middleware/jwt_auth.go

```diff
@@ -5,58 +5,72 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
-	"github.com/dgrijalva/jwt-go"
 	"log"
 	"net/http"
 	"opg-file-service/internal"
 	"strings"
+
+	"github.com/dgrijalva/jwt-go"
 )
 
 type HashedEmail struct{}
 
-func JwtVerify(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
-		jwtSecret := internal.GetEnvVar("JWT_SECRET", "MyTestSecret")
+type Cacheable interface {
+	GetSecretString(key string) (string, error)
+}
 
-		//Get the token from the header
-		header := r.Header.Get("Authorization")
+func JwtVerify(secretsCache Cacheable) func(next http.Handler) http.Handler {
```

Comment:
Didn't know you could do a dual function declaration like this?

Reply:
This is a functional programming concept known as Currying, where a function takes a parameter and returns a function (that may also take a parameter and return a function, and so on). In this instance, the `JwtVerify` is a handler function that needs to return a `http.Handler` but I needed to initialise it with a cache. The OOP option would be something like:

```
type JwtVerifier struct {
    cache Cacheable
}

func (v *JwtVerifier) Verify(next http.Handler) http.Handler {
...
}

// and then call it in main like
jwtVerifier := middleware.JwtVerifier{cache}
getRouter.Use(jwtVerifier.Verify)

```

Here's a more thorough explanation using Golang, though it is easier and cleaner to do this in more functional languages: https://blog.usejournal.com/function-currying-in-go-a88672d6ebcf